src/promptflow/promptflow/executor/_tool_resolver.py

```diff
@@ -129,15 +129,16 @@ def resolve_tool_by_node(self, node: Node, convert_input_types=True) -> Resolved
 
     def _load_source_content(self, node: Node) -> str:
         source = node.source
-        if source is None or source.path is None or not Path(self._working_dir / source.path).exists():
+        # If is_file returns True, the path points to a existing file, so we don't need to check if exists.
+        if source is None or source.path is None or not Path(self._working_dir / source.path).is_file():
             raise InvalidSource(
                 target=ErrorTarget.EXECUTOR,
                 message_format="Node source path '{source_path}' is invalid on node '{node_name}'.",
                 source_path=source.path if source is not None else None,
                 node_name=node.name,
             )
-        with open(self._working_dir / source.path) as fin:
-            return fin.read()
+        file = Path(self._working_dir / source.path)
```

Comment:
Can be simplified to `file = self._working_dir / source.path`? As long as the `/` operator works, we can confirm that `self._working_dir` is already a `Path` object.